evmbin/src/info.rs

```diff
@@ -100,51 +100,58 @@ pub fn run_action<T: Informant>(
 	})
 }
 
+/// Input data to run transaction
+#[derive(Debug)]
+pub struct TxInput<'a, T> {
+	/// Chain specification name associated with the transaction.
+	pub name: &'a String,
+	/// Transaction index from list of transactions within a state root hashes corresponding to a chain.
+	pub idx: usize,
+	/// Fork specification (i.e. Constantinople, EIP150, EIP158, etc).
+	pub spec: &'a ethjson::spec::ForkSpec,
+	/// State of all accounts in the system that is a binary tree mapping of each account address to account data
+	/// that is expressed as Plain Old Data containing the account balance, account nonce, account code in bytes,
+	/// and the account storage binary tree map.
+	pub pre_state: &'a pod_state::PodState,
+	/// State root hash associated with the transaction.
+	pub post_root: H256,
+	/// Client environment information associated with the transaction's chain specification.
+	pub env_info: &'a client::EnvInfo,
+	pub transaction: transaction::SignedTransaction,
+	/// JSON formatting informant.
+	pub informant: T,
+	pub trie_spec: &'a TrieSpec,
+}
+
 /// Execute given transaction and verify resulting state root.
 pub fn run_transaction<T: Informant>(
-	// Chain specification name associated with the transaction.
-	name: &str,
-	// Transaction index from list of transactions within a state root hashes corresponding to a chain.
-	idx: usize,
-	// Fork specification (i.e. Constantinople, EIP150, EIP158, etc).
-	spec: &ethjson::spec::ForkSpec,
-	// State of all accounts in the system that is a binary tree mapping of each account address to account data
-	// that is expressed as Plain Old Data containing the account balance, account nonce, account code in bytes,
-	// and the account storage binary tree map.
-	pre_state: &pod_state::PodState,
-	// State root hash associated with the transaction.
-	post_root: H256,
-	// Client environment information associated with the transaction's chain specification.
-	env_info: &client::EnvInfo,
-	transaction: transaction::SignedTransaction,
-	// JSON formatting informant.
-	mut informant: T,
-	trie_spec: TrieSpec,
+	tx_input: TxInput<T>
 ) {
-	let spec_name = format!("{:?}", spec).to_lowercase();
-	let spec = match EvmTestClient::spec_from_json(spec) {
+	let mut tx_input = tx_input;
```

Comment:
```suggestion
```
This line is redundant, cause you can mark it `mut` in the signature.

Reply:
The cool thing you can also do to avoid rewriting the rest of the code is to destructure `tx_input`:
```
let TxInput { idx, spec, pre_state, post_root, informant, ... } = tx_input;
```